posts/posts-router.js

```diff
@@ -0,0 +1,91 @@
+const express = require('express');
+
+const db = require('../data/db.js');
+
+const router = express.Router();
+
+//Get
+router.get('/', (req, res) => {
+    db
+        .find()
+        .then(post => {
+            res.status(200).json(post);
+        })
+        .catch(err => {
+            res.status(500).json({ error: 'There was an error while saving the post to the database.' });
+        });
+});
+
+//Get(post)
+router.get('/:id', (req, res) => {
+    const id = req.params.id;
+    db
+        .findById(id)
+        .then(post => {
+            if(!user) {
```

Comment:
```suggestion
            if(!post) {
```
user is not defined so it go to the .catch automatically and that's why it doesn't work